pandas/tests/indexes/datetimes/test_arithmetic.py

```diff
@@ -44,6 +44,23 @@ def addend(request):
 
 
 class TestDatetimeIndexComparisons(object):
+    @pytest.mark.parametrize('other', [None,
+                                       datetime(2016, 1, 1).date()])
+    def test_dti_cmp_invalid(self, tz, other):
+        # GH#19301
+        dti = pd.date_range('2016-01-01', periods=2, tz=tz)
+
+        assert not (dti == other).any()
+        assert (dti != other).all()
+        with pytest.raises(TypeError):
+            dti < other
+        with pytest.raises(TypeError):
+            dti <= other
+        with pytest.raises(TypeError):
+            dti > other
+        with pytest.raises(TypeError):
+            dti >= other
+
     # TODO: De-duplicate with test_comparisons_nat below
```

Comment:
is there a reason NOT to de-dupe with the other comparisons in this file (e.g. the TODO comment), in this PR? This is adding code then then will be inserted into the parameterization, I would rather just do it right once.

Reply:
I think this comment was added in #19317, which was largely intended to be cut/paste with edits done in a separate PR.  I can go and add this de-duplication to this PR, but that would be expanding the scope.

Reply:
let's do it. right now these are a bit orphaned and should be combined with existing tests. in general this is a good thing to do anyhow.

Reply:
Done.  Turned out they weren't duplicated so much as poorly-named.